### DISCUSSION TITLE
如何知道int與float等資料型別的變數所佔的RAM大小？

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#214] [Q&A] [answered]
我一直搞不懂，int與float等資料型別的變數，所佔的RAM的大小是去哪裡看的？因為Python有分int與float，但[https://www.python.org/](https://www.python.org/)裡面好像沒有說明。

- int的全名是integer，中文是「整數」，也就是除以1餘數為0的數=被1整除的數
- float的全名就是float，中文是「浮點數」。更精確講是「單精度浮點數」，我目前的理解是float就是有小數點的數，如2.=2.0。

**Comment by xCk27x** [comment, marked as answer]
可以用sys函數庫裡的 getsizeof() 函數取得
import sys就可使用

例如sys.getsizeof(123)得到28
因為大於0且小於1073741824的int占用28個bytes
而sys.getsizeof(1.23)得到24
因為float占用24個bytes

剩下的資料型態共佔多少個bytes可以參考下面的連結
[python-sys.getsizeof的使用](https://blog.csdn.net/qm5132/article/details/100557950)

**Reply by enoch-lightning** [reply]
謝謝你的回答，我來研究看看。
